src/InputWithOptions/InputWithOptions.js

```diff
@@ -96,6 +96,15 @@ class InputWithOptions extends Component {
     ) {
       this.showOptions();
     }
+
+    // Clear value in controlled mode
+    if (
+      prevProps.value !== '' &&
+      this.props.value === '' &&
+      this.state.inputValue !== ''
```

Comment:
Isn't it be enough just to check if the new `value` prop is an empty string?
```suggestion
      prevProps.value !== this.props.value && this.props.value == ''
```